src/Context.php

```diff
@@ -34,9 +37,8 @@ public function __construct($args)
 
         // algorithm for signing; not necessary for verifying.
         if (isset($args['algorithm'])) {
-            $this->algorithmName = $args['algorithm'];
+            $this->algorithm = Algorithm::create($args['algorithm']);
```

Comment:
Any reason for this change?

Reply:
This old parameter only handled the HMAC case, where a singing and verifying key are the same. Needed a class to discriminate HMAC from RSA, so invoke with a factory that makes the right underlying Algorithm subclass.